src/test/java/uk/gov/companieshouse/api/accounts/controller/PreviousPeriodControllerTest.java

```diff
@@ -0,0 +1,110 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class PreviousPeriodControllerTest {
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private PreviousPeriod previousPeriod;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private CompanyAccountEntity companyAccountEntity;
+
+    @Mock
+    private PreviousPeriodService previousPeriodService;
+
+    @Mock
+    private ApiResponseMapper apiResponseMapper;
+
+    @Mock
+    private SmallFull smallFull;
+
+    @Mock
+    private Map<String, String> links;
+
+    @InjectMocks
+    private PreviousPeriodController previousPeriodController;
+
+    @BeforeEach
+    public void setUp() throws NoSuchAlgorithmException, DataException {
+        when(request.getAttribute("transaction")).thenReturn(transaction);
+        when(request.getHeader("X-Request-Id")).thenReturn("test");
+        ResponseObject responseObject = new ResponseObject(ResponseStatus.CREATED,
+            previousPeriod);
+        doReturn(responseObject).when(previousPeriodService)
+            .create(any(PreviousPeriod.class), any(Transaction.class), anyString(), anyString());
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.CREATED)
+            .body(responseObject.getData());
+        when(apiResponseMapper.map(responseObject.getStatus(),
+            responseObject.getData(), responseObject.getValidationErrorData()))
+            .thenReturn(responseEntity);
+        doReturn(transaction).when(request)
+            .getAttribute(AttributeName.TRANSACTION.getValue());
+        doReturn(smallFull).when(request).getAttribute(AttributeName.SMALLFULL.getValue());
+        doReturn(companyAccountEntity).when(request)
+            .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+        doReturn("12345").when(companyAccountEntity).getId();
+        doReturn(responseObject).when(previousPeriodService).findById("create", "test");
+        doReturn(new ResponseObject(ResponseStatus.FOUND,
+            previousPeriod)).when(previousPeriodService).findById("find", "test");
+        doReturn("123456").when(transaction).getCompanyNumber();
+        doReturn(links).when(smallFull).getLinks();
+        doReturn("7890").when(links).get("self");
+    }
+
+    @Test
+    @DisplayName("Tests the successful creation of a previous period resource")
+    public void canCreatePreviousPeriod() throws NoSuchAlgorithmException {
+        ResponseEntity response = previousPeriodController
+            .create(previousPeriod, request);
+        assertNotNull(response);
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertEquals(previousPeriod, response.getBody());
```

Comment:
Is it worth verifying that each of the mocked stubs have been called in here? Use of Mockito's `verify` will assert that a method has been called with expected parameters a certain number of `times`.